minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/configuration/notifiers/FileChangeNotifier.java

```diff
@@ -63,15 +71,23 @@
     }
 
     @Override
-    public void notifyListeners() {
+    public ListenerHandleResult[] notifyListeners() {
+        logger.info("Notifying Listeners of a change");
         final File fileToRead = configFile.toFile();
+
+        ListenerHandleResult[] listenerHandleResults = new ListenerHandleResult[configurationChangeListeners.size()];
+        int count = 0;
         for (final ConfigurationChangeListener listener : getChangeListeners()) {
             try (final FileInputStream fis = new FileInputStream(fileToRead);) {
                 listener.handleChange(fis);
-            } catch (IOException ex) {
-                throw new IllegalStateException("Unable to read the changed file " + configFile, ex);
+                listenerHandleResults[count] = new ListenerHandleResult(listener);
+            } catch (IOException | ConfigurationChangeException ex) {
+                listenerHandleResults[count] = new ListenerHandleResult(listener, ex);
             }
+            logger.info("Listener notification result:" + listenerHandleResults[count].toString());
+            count ++;
         }
+        return new ListenerHandleResult[0];
```

Comment:
this should be listenerHandleResults.  this is only returning a new empty array.